src/reconcile/rec_visibility.c

```diff
@@ -265,12 +265,27 @@ static int
 __rec_find_and_save_delete_hs_upd(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_INSERT *ins,
   WT_ROW *rip, WTI_UPDATE_SELECT *upd_select)
 {
-    WT_UPDATE *delete_tombstone, *delete_upd;
+    WT_UPDATE *delete_tombstone, *delete_upd, *triggering_prepare_upd;
+    uint8_t prepare_state;
+    bool find_triggering_prepare;
 
     delete_tombstone = NULL;
-
-    for (delete_upd = upd_select->tombstone != NULL ? upd_select->tombstone : upd_select->upd;
+    find_triggering_prepare = F_ISSET(S2C(session), WT_CONN_PRESERVE_PREPARED) &&
+      WT_TIME_WINDOW_HAS_START_PREPARE(&upd_select->tw);
+    /*
+     * If we're in preserve_prepared mode and have a prepared time window, look for the prepared
+     * update that triggered the history store deletion. This helps us determine if we should delay
+     * the deletion until the prepared update is committed/rolled back and stable.
+     */
+    for (delete_upd = upd_select->tombstone != NULL ? upd_select->tombstone : upd_select->upd,
+        triggering_prepare_upd = NULL;
          delete_upd != NULL; delete_upd = delete_upd->next) {
+        if (find_triggering_prepare && delete_upd->type != WT_UPDATE_RESERVE) {
+            WT_ACQUIRE_READ(prepare_state, delete_upd->prepare_state);
+            if (prepare_state != WT_PREPARE_INIT)
+                triggering_prepare_upd = delete_upd;
```

Comment:
Can you add a comment here that the `triggering_prepare_upd` points to the last prepared update in the chain.